eclair-core/src/main/scala/fr/acinq/eclair/router/BalanceEstimate.scala

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.router
+
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.bitcoin.{LexicographicalOrdering, Satoshi}
+import fr.acinq.eclair.MilliSatoshi.toMilliSatoshi
+import fr.acinq.eclair.router.Graph.GraphStructure.{DirectedGraph, GraphEdge}
+import fr.acinq.eclair.router.Router.{ChannelHop, PublicChannel}
+import fr.acinq.eclair.{MilliSatoshi, TimestampSecond}
+
+import scala.concurrent.duration.FiniteDuration
+
+/** Estimates the balance between a pair of nodes
+ *
+ * @param low           lower bound on the balance
```

Comment:
Really like the naming `low` and `high` might overtake this. I used `minLiquidity` and `maxLiquidity` which is a bit more expressive but always a bit of a burden to carry around.